service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageNotificationEmailService.kt

```diff
@@ -100,36 +81,27 @@ class BulletinNotificationEmailService(
         }
     }
 
-    fun sendBulletinNotification(db: Database, msg: SendUnreadBulletinNotificationEmail) {
+    fun sendMessageNotification(db: Database, msg: SendMessageNotificationEmail) {
+        val (messageRecipientId, personEmail, language) = msg
+
         db.transaction { tx ->
             emailClient.sendEmail(
-                traceId = msg.id.toString(),
-                toAddress = msg.receiverEmail,
-                fromAddress = getFromAddress(msg.language),
-                getSubject(msg.language),
-                getHtml(msg.language),
-                getText(msg.language)
-            )
-
-            // Mark as sent
-            tx.createUpdate(
-                """
-                UPDATE bulletin_instance 
-                SET notification_sent_at = :sent_at
-                WHERE id = :id
-                """.trimIndent()
+                traceId = messageRecipientId.toString(),
+                toAddress = personEmail,
+                fromAddress = getFromAddress(language),
+                subject = getSubject(language),
+                htmlBody = getHtml(language),
+                textBody = getText(language)
             )
-                .bind("id", msg.id)
-                .bind("sent_at", OffsetDateTime.now())
-                .execute()
+            tx.markNotificationAsSent(messageRecipientId)
         }
     }
 
     private fun getSubject(language: Language): String {
         val postfix = if (System.getenv("VOLTTI_ENV") == "prod") "" else " [${System.getenv("VOLTTI_ENV")}]"
 
         return when (language) {
-            Language.en -> "New bulletin in eVaka$postfix"
+            Language.en -> "New Message in eVaka$postfix"
```

Comment:
```suggestion
            Language.en -> "New message in eVaka$postfix"
```

Reply:
Actually all the subject versions should be fixed now that we have messages too.

Reply:
I kept the subjects shorter, bulletin is mentioned in the message body